master/static/srv/proto_get_trial_ids_for_experiment.sql

```diff
@@ -0,0 +1,66 @@
+WITH searcher_info AS (
+    SELECT config->'searcher'->>'metric' AS metric_name,
+           (
+               SELECT CASE
+                          WHEN coalesce(
+                                  (
+                                              config->'searcher'->>'smaller_is_better'
+                                      )::boolean,
+                                  true
+                              ) THEN 1
+                          ELSE -1
+                          END
+           ) AS sign
+    FROM experiments e
+    WHERE e.id = $1
+), filtered_experiment_trials AS (
+    SELECT
+        t.id AS id,
+        'STATE_' || t.state AS state,
+        t.start_time,
+        t.end_time,
+        coalesce(t.end_time, now()) - t.start_time AS duration,
+        (
+            SELECT s.prior_batches_processed + s.num_batches
+            FROM steps s
+            WHERE s.trial_id = t.id
+                AND s.state = 'COMPLETED'
+            ORDER BY s.id DESC
+            LIMIT 1
+        ) AS total_batches_processed,
+        (
+           CASE WHEN t.best_validation_id IS NOT NULL THEN
+                (SELECT searcher_info.sign * (v.metrics->'validation_metrics'->>searcher_info.metric_name)::float8
+                FROM validations v
+                WHERE v.id = t.best_validation_id
+                LIMIT 1)
+            ELSE
+                -- For trials before `public.trials.best_validation_id` was added.
+                (SELECT searcher_info.sign * (v.metrics->'validation_metrics'->>searcher_info.metric_name)::float8
+                 FROM validations v
+                 WHERE v.trial_id = t.id
+                 ORDER BY searcher_info.sign * (v.metrics->'validation_metrics'->>searcher_info.metric_name)::float8 ASC
+                 LIMIT 1)
+            END
+        ) as best_signed_search_metric,
+        (
+           SELECT v.end_time
+           FROM validations v
+           WHERE v.end_time IS NOT NULL
+           ORDER BY v.id DESC
+           LIMIT 1
+        ) as validation_end_time
+    FROM trials t, searcher_info
+    WHERE t.experiment_id = $1
+      AND ($2 = '' OR t.state IN (SELECT unnest(string_to_array($2, ','))::trial_state))
+), page_info AS (
+    SELECT public.page_info((SELECT COUNT(*) AS count FROM filtered_experiment_trials), $3, $4) AS page_info
+)
+SELECT
+    (SELECT coalesce(json_agg(paginated_experiment_trials), '[]'::json) FROM (
+        SELECT id FROM filtered_experiment_trials
+        ORDER BY %s
+        OFFSET (SELECT p.page_info->>'start_index' FROM page_info p)::bigint
```

Comment:
nonblocking: Why not just use `$3` or `$4` for OFFSET and LIMIT?

Reply:
If you just use `$4` for limit then if it's zero we dont default it. latest master `GET /api/v1/experiments` with no params is returning `[]` which is a bug. Offset has some similar problems. Also this is partially to keep the same pagination behavior as before.

Reply:
Really, you could default limit outside and just use them - the other reason I was avoid that was it was nice this way to return the full `apiv1.Pagination` json so we can just `QueryProtof` into the resp object.